src/appleseed/foundation/utility/iesparser.cpp

```diff
@@ -0,0 +1,946 @@
+                line_counter);
+        }
+    }
+    if (!check_increasing_order(horizontal_angles))
+    {
+        throw ParsingException("Horizontal angles"
+            " must be mentioned in increasing order", line_counter);
+    }
+}
+
+
+bool IESParser::is_keyword_line(const std::string& line)
+{
+    boost::regex regex(KEYWORD_LINE_REGEX);
+    return boost::regex_match(line, regex);
+}
+
+
+bool IESParser::is_tilt_line(const std::string& line)
+{
+    boost::regex regex(TILT_LINE_REGEX);
+    return boost::regex_match(line, regex);
+}
+
+
+void IESParser::parse_keyword_line(const std::string& line)
+{
+    boost::regex regex(KEYWORD_LINE_REGEX);
+    boost::smatch what;
+    if (!boost::regex_match(line, what, regex))
+    {
+        throw ParsingException("Keyword is expected", line_counter);
+    }
+
+    std::string key = what[1];
+    std::string value = what[2];
+
+    // Check if the specified standard allows this keyword.
+    if (!ignore_allowed_keywords) accept_keyword(key);
+
+    // Process MORE, BLOCK and ENDBLOCK keywords separately.
+    // For all other keywords - just add them to dictionary.
+    process_block_keyword(key);
+    if (key == "MORE")
+    {
+        if (last_added_keyword == keywords_dictionary.end())
+        {
+            throw ParsingException(
+                "Keyword MORE occured before any other keyword", line_counter);
+        }
+        last_added_keyword->second += ("\n" + value);
+    }
+    else
+    {
+        KeywordsDictionary::iterator it = keywords_dictionary.find(key);
+        if (it != keywords_dictionary.end())
+        {
+            throw ParsingException(
+                ("Keyword " + key + " is duplicated").c_str(), line_counter);
+        }
+        keywords_dictionary[key] = value;
+        last_added_keyword = keywords_dictionary.find(key);
+    }
+}
+
+
+void IESParser::parse_tilt_line(const std::string& line)
+{
+    boost::regex regex(TILT_LINE_REGEX);
+    boost::smatch what;
+    if (!boost::regex_match(line, what, regex))
+    {
+        throw ParsingException("TILT line is expected", line_counter);
+    }
+
+    std::string value = what[1];
+
+    if (value == "INCLUDE")
+        tilt_specification = IncludeTilt;
+    else if (value == "NONE")
+        tilt_specification = NoTilt;
+    else if (!ignore_tilt)
+    {
+        throw NotSupportedException(
+            "TILT specification from file is not supported", line_counter);
+    }
+}
+
+
+void IESParser::process_block_keyword(const std::string& keyword)
+{
+    if (keyword == "ENDBLOCK" || keyword == "BLOCK")
+    {
+        throw NotSupportedException("Block keywords are not supported", line_counter);
+    }
+}
+
+
+void IESParser::accept_keyword(const std::string& keyword)
+{
+    if (keyword.empty())
+    {
+        throw ParsingException("Keyword is empty", line_counter);
+    }
+
+    assert(format != UnknownFormat);
+
+    switch (format)
+    {
+    case Format2002:
```

Comment:
`case`, `default`, `private`, `protected` and `public` keywords are indented with two spaces. Like everything else 
they deserve an indentation, but not a full one :)